content/en/blog/news/2021-moving-v2-infra.md

```diff
@@ -109,3 +112,18 @@ Have we missed anything?  Please contribute to this blog or feedback on the slac
 ## Why are only the most recent versions v2 images copied to GitHub packages and not all versions?
  
 There are 14 Terabytes of data that make up the jenkinsxio container registry on GCP, it would be costly and wasteful to transfer all this to GitHub so we picked the last known version of each image that was released last year.  If there are specific images that you wish to use either pull / push them yourself to a container registry of your own or reach out and on a case by case effort, we can look to move them to GitHub while the read permissions are made private and before the project is shut down.
+
+## I'm on v2 and use a builder image which is not available on GitHub container registry, how do I build my own version to work with the new helm and image repositories?
+
+The old v2 jx code [lives on a branch](https://github.com/jenkins-x/jx/tree/v2) you will need to:
+
+- fork this branch
+- apply search and replace changes for 
+  i) gcr.io/jenkinsxio to ghcr.io/jenkins-x [example](https://github.com/jenkins-x/jx/commit/9d5f1aa6421a22e311d12482893586a45b485ac8)
+  ii) https://storage.googleapis.com/chartmuseum.jenkins-x.io to https://jenkins-x-charts.github.io/v2 [example](https://github.com/jenkins-x/jx/commit/88965b1696c81349f4e330aeb98ec5e26116341c)
+- run `make linux` to build the updated jx binary
+- build and push the images you require, for example
+```
+docker build -f Dockerfile.builder-maven -t ghcr.io/jenkins-x/builder-maven:2.1.149-768-patch2 .
+docker push ghcr.io/jenkins-x/builder-maven:2.1.149-768-patch2
+```
```

Comment:
Something you can add here is that once all builder-jx/go/maven/etc.. are pushed, they will need to host the charts
- download each needed chart/subchart needed to make jx work, exemple for [jenkins-x-platform/](https://github.com/jenkins-x/jenkins-x-platform/releases ), [jxboot-resources/releases](https://github.com/jenkins-x-charts/jxboot-resources/releases)
- on each charts/subcharts replace current `gcr.io/jenkinsxio` images with `ghcr.io/jenkins-x` if it exists or rebuild and host them on a private registry if not
  - not sure if it's possible to override subchart value from the parent chart in the gitops repository path `env/values.tmpl.yaml` directly with the new image values
- host on a private chartmuseum the needed and edited charts/subcharts with new images to make jx work

Reply:
great call @JordanGoasdoue - lemme add those steps in